packages/editor/src/store/effects/posts.js

```diff
@@ -166,6 +167,32 @@ export const requestPostUpdate = async ( action, store ) => {
 	}
 };
 
+/**
+ * Update view as link handler.
+ * Performs a side effect of updating the "View as" link on the top bar.
+ * This link is generated on the server and during a post update it may change,
+ * and the page is not reloaded so the link continues to be the same,
+ * this effect makes sure this link is always up to date.
+ *
+ * @param {Object} action  action object.
+ * @param {Object} store   Redux Store.
+ */
+export const updateViewAsLink = ( action ) => {
+	const newPermalink = action.post.link;
+	const previousPermalink = action.previousPost.link;
+	if ( newPermalink === previousPermalink || ! newPermalink ) {
+		return;
+	}
+
+	const nodeToUpdate = document.querySelector(
+		`#${ VIEW_AS_CONTAINER_ID } a[href="${ previousPermalink }"]`
+	);
+	if ( ! nodeToUpdate ) {
+		return;
+	}
+	nodeToUpdate.setAttribute( 'href', newPermalink );
+};
```

Comment:
Can we move this effect to the `edit-post` module instead? Ideally, the `editor` package has no knowledge about any WP context.